pkg/apis/batch/validation/validation.go

```diff
@@ -565,6 +567,35 @@ func ValidateJobTemplateSpec(spec *batch.JobTemplateSpec, fldPath *field.Path, o
 	return allErrs
 }
 
+func validateCompletions(spec, oldSpec batch.JobSpec, fldPath *field.Path) field.ErrorList {
+	if !feature.DefaultFeatureGate.Enabled(features.ElasticIndexedJob) {
+		return apivalidation.ValidateImmutableField(spec.Completions, oldSpec.Completions, fldPath)
+	}
```

Comment:
rather than checking feature enablement inside validation.go, plumb whether to allow mutable completions in a new bool field in `JobValidationOptions`, set in `validationOptionsForJob`

(the way this was checking feature gates was not wrong, but most feature gate checks in validation.go risk not properly tolerating existing data, so we try to avoid importing feature gates in those files at all)